src/dynamics/video_player/player/player.js

```diff
@@ -1286,45 +1294,58 @@ Scoped.define("module:VideoPlayer.Dynamics.Player", [
+                    this.imgEle.style.width = '100%';
+                    this.vidEle.setAttribute('crossorigin', 'anonymous')
+                    this.vidEle.muted = true;
+                    this.set('trackFrameTime', 0);
 
-                    img = document.createElement('img');
-                    img.style.width = '100%';
-                    const imgElements = videoParentEle.querySelectorAll('img');
-
-                    const vidEle = document.createElement('video');
-                    vidEle.src = this.get("source");
-                    vidEle.setAttribute('crossorigin', 'anonymous')
-                    vidEle.muted = true;
-                    vidEle.play();
-
-
+                    this.vidEle.addEventListener("loadeddata", (event) => {
+                        this.vidEle.pause();
+                        this._drawFrame(this.vidEle, currentPosition, video.videoWidth, video.videoHeight, (canvas) => {
+                            this.removeImgElements(videoParentEle.querySelectorAll('img'));
+                            this.imgEle.src = `${canvas.toDataURL()}`;
+                            videoParentEle.appendChild(this.imgEle);
 
+                        })
+                    })
+                },
+                _renderVideoFrame: function(video, isEnded) {
+                    const currentPosition = this.getCurrentPosition();
+                    video.style.backgroundColor = 'transparent';
+                    const videoParentEle = video.parentElement;
 
                     try {
-                        vidEle.addEventListener("loadeddata", (event) => {
-                            this._drawFrame(vidEle, currentPosition, video.videoWidth, video.videoHeight, (canvas) => {
+                        this.vidEle.pause();
+                        this.imgEle.src = '';
+                        this._drawFrame(this.vidEle, currentPosition, video.videoWidth, video.videoHeight, (canvas) => {
+                            this.removeImgElements(videoParentEle.querySelectorAll('img'));
+                            this.imgEle.src = `${canvas.toDataURL()}`;
+                            videoParentEle.appendChild(this.imgEle);
 
-                                imgElements.forEach(img => {
-                                    if (img.parentNode)
-                                        img.parentNode.removeChild(img)
-                                })
 
-                                img.src = `${canvas.toDataURL()}`;
-                                videoParentEle.appendChild(img);
-
-                                if (this.get("trackFrameTime") > currentPosition) {
+                            if (this.get("trackFrameTime") > currentPosition || isEnded) {
+                                if(isEnded){
+                                    video.currentTime = 0;
+                                } else {
                                     video.currentTime = this.get("trackFrameTime");
                                 }
```

Comment:
Hi @maxblessed, this is updating the actual video's position when the frame is rendered. If I rewind the video and pause it, for example, the video moves back to the position before I rewound it.